cuqi/implicitprior/_denoiser.py

```diff
@@ -0,0 +1,108 @@
+#%%
+from abc import ABC, abstractmethod
+from cuqi.distribution import Distribution
+import numpy as np
+
+class DenoiseRegularizer(Distribution):
+    """    
+    This class defines implicit regularized priors for which we can apply 
+    gradient-based algorithms (ex:MYULA). The regularization is performed using
+    a denoising algorithm as we can encounter in MYULA
+    and PnP-ULA.
+    
+    Ex: Moreau-Yoshida based regularization
```

Comment:
What does "Ex:" signify here, "example", please elaborate.  Also, is the Moreau-Yoshida regularization precisely what this class can do, or is it just an example? If just an example, I think it should be clarified that this function is more general.

Reply:
It was a bit missleading. We updated the documentation to make it clearer.

Reply:
Status update: as @remilaumont pointed out, the documentation is updated